services/navigation/builtin/builtin.go

```diff
@@ -280,9 +280,15 @@ func (svc *builtIn) SetMode(ctx context.Context, mode navigation.Mode, extra map
 	cancelCtx, cancelFunc := context.WithCancel(context.Background())
 	svc.wholeServiceCancelFunc = cancelFunc
 	svc.mode = mode
-	if svc.mode == navigation.ModeWaypoint {
+
+	switch svc.mode {
+	case navigation.ModeManual:
+	case navigation.ModeWaypoint:
 		svc.startWaypointMode(cancelCtx, extra)
+	case navigation.ModeExplore:
+		return errors.New("navigation mode 'explore' is not currently available")
```

Comment:
[note] Do we want to add in a case for an unknown/unspecified navigation mode to return an error? This did not exist in the previous version of `SetMode`.

Reply:
i think so!

Reply:
yeah, i think that makes sense

Reply:
My mistake, since we define specific consts of navigation modes, there is no way an unknown mode gets past the client/server. Tests for unknown modes being sent to each have been added.